lesson08/index.js

```diff
@@ -0,0 +1,42 @@
+"use strict";
+
+function createListDom(itemList) {
+    const ulElement = document.getElementById("ul");
+    const frag = document.createDocumentFragment();
+
+    itemList.forEach(function(item) {
+        const liElement = document.createElement("li");
+        const anchorElement = document.createElement("a");
+        const imageElement = document.createElement("img");
+        anchorElement.href= `/${item.to}`;
+        anchorElement.textContent = item.text;
+        imageElement.src = item.img;
+        imageElement.alt = item.alt;
+        frag.appendChild(liElement).appendChild(anchorElement).appendChild(imageElement)
+    });
+    ulElement.appendChild(frag);
+}
+
+function createLoading() {
+    const bodyElement = document.querySelector('.body');
+    bodyElement.insertAdjacentHTML('afterbegin', '<div class="loading"><img class="loading-img" src="./img/loading-circle.gif"></div></div>');
```

Comment:
IMO
forget above comment. i misunderstood.

how about this
```js
function createLoading() {
  const img = document.createElement("img");
  img.src = "./img/loading-circle.gif";
  img.id = "js-img"
  ulElement.appendChild(img);
}

// function removeLoading() {
//   ulElement.removeChild(document.querySelector("#js-img"));
// }
```

Reply:
I think it is impossible to insert the img tag as a child element of ul according to the rules.
https://yoshikawaweb.com/element/

Does it mean that the elements are generated one by one instead of being inserted all at once in html because of their high extensibility, maintainability, and readability?

Reply:
> I think it is impossible to insert the img tag as a child element of ul according to the rules.
https://yoshikawaweb.com/element/
So I thought it would be better to create a div and generate it outside the ul element.

surely. your right. 

> Does it mean that the elements are generated one by one instead of being inserted all at once in html because of their high extensibility, maintainability, and readability?

yes.
1. AppendChild is better from a [performance point of view](https://www.measurethat.net/Benchmarks/Show/11738/0/insertadjacenthtml-vs-append)

Reply:
Rewritten with appendChild and before.

cd28586

https://codesandbox.io/s/modest-khorana-yicng?file=/index.js

There is a measurement tool called MeasureThat.net! I did not know that.